### DISCUSSION TITLE
Explore: Format Log Labels after json/logfmt parsing a Loki Result

### DISCUSSION THREAD

**Discussion by edthamm** [grafana/grafana#42935] [Observability]
**Story:**

As Software Engineer
I want to see well formatted output in the log labels section of Explore
so that I can easily parse the content and debug faster.

---------

**Background:**

Lets say I have this query
```
{service="my-service"}
|= "error"
|= "stacktrace"
| json
| line_format "{{.message}}"
```

The result is exactly what I would expect and also want:
![image](https://user-images.githubusercontent.com/758377/145412337-310128ee-0848-46bb-b10a-aec12c93b892.png)
which is amazing for browsing.

However if I then expand, the entry the content of the extracted labels is not formatted nicely but rather represented as one big blob of text like this:
![image](https://user-images.githubusercontent.com/758377/145412805-a831a30a-a393-4ccf-bf2f-50656f5b5e04.png)

It would be pretty nice if it where formatted like it is when using `line_format "{{.stacktrace}}"`. So basically like this:
![image](https://user-images.githubusercontent.com/758377/145413040-0a24b8a3-93c8-4a2e-8af6-b32ca752e9a9.png)

Note this is the exact same log entry I use throughout all my examples.

Here is a more complete view of the log entry to provide some context to the specific snippets above:
![image](https://user-images.githubusercontent.com/758377/145414157-e664f51b-464c-4753-b304-eb3065301642.png)

A nicer formatting in the log labels context would help a lot in debugging as we could use the line formatted display for quick identification of relevant lines and then use the log lines to start digging into the issues.
Please reach out if you need more details or information and thanks for the amazing product you people create.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by POnakS** [comment]
@ed

I see this button - after pressing this, my logs are correclty formated (you may also need to check wrap lines)
![image](https://github.com/grafana/grafana/assets/90155580/0a29631c-5f48-4b88-b74e-be488b95b581)
